v2/tools/generator/internal/config/group_configuration.go

```diff
@@ -91,10 +108,25 @@ func (gc *GroupConfiguration) Add(version *VersionConfiguration) *GroupConfigura
 }
 
 // findVersion uses the provided TypeName to work out which nested VersionConfiguration should be used
-func (gc *GroupConfiguration) findVersion(name astmodel.TypeName) (*VersionConfiguration, bool) {
-	v := strings.ToLower(name.PackageReference.PackageName())
-	version, ok := gc.versions[v]
-	return version, ok
+func (gc *GroupConfiguration) findVersion(name astmodel.TypeName) (*VersionConfiguration, error) {
+	ref := name.PackageReference
+	if s, ok := ref.(astmodel.StoragePackageReference); ok {
+		// If we have a storage package reference, need to unwrap the actual local package reference
+		// as all our configuration is based on API versions, not storage versions
+		ref = s.Local()
+	}
+
+	v := strings.ToLower(ref.PackageName())
+	if version, ok := gc.versions[v]; ok {
+		return version, nil
+	}
+
+	return nil, errors.Errorf(
+		"configuration of group %s has no detail for version %s (configured versions are: %s)",
+		gc.name,
+		ref.PackageName(),
+		strings.Join(gc.configuredVersions(), ", "))
```

Comment:
If you're a bit worried about the performance impact of doing this repeatedly, I think the name and package name that it's trying to look up are the most useful parts here. The configured versions are easy enough to see in the file, it's the name of the type (as it is in this specific part of the pipeline between transformations) that's hard to determine when running the generator.

Reply:
I found the list of options very useful when debugging my WIP branch. What I've done with the custom error type is to defer the sort/join until the error is rendered, which mitigates much of the cost.